File: Tan Khoa/cses2428.py
from collections import defaultdict

n, k = map(int, input().split())
arr = list(map(int, input().split()))

left, right = 0, 0
ans = 0

cnt = defaultdict(lambda: 0)
exist = dict()

while right < n:
	if len(exist) + 1 <= k or arr[right] in exist:
		cnt[arr[right]] += 1
		exist[arr[right]] = True
		ans += right - left + 1
		right += 1
	else:
		cnt[arr[left]] -= 1
		if cnt[arr[left]] == 0:
			exist.pop(arr[left])
		left += 1


print(ans)

File: Tan Khoa/cses2428.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;
#define nln '\n'

int main() {
	cin.tie(0)->sync_with_stdio(0);

	ll n, k;	cin >> n >> k;
	vector<ll> arr(n);
	for (auto &v : arr)
		cin >> v;

	map<ll, ll> cnt;
	map<ll, bool> exist;

	ll l = 0, r = 0, ans = 0;
	while (r < n) {
		if (exist.size() + 1 <= k || exist.find(arr[r]) != exist.end()) {
			cnt[arr[r]]++;
			exist[arr[r]] = true;
			ans += r-l+1;
			r++;
		} else {
			cnt[arr[l]]--;
			if (cnt[arr[l]] == 0)
				exist.erase(arr[l]);
			l++;
		}
	}

	cout << ans << nln;

	return 0;
}